hashintel/hash
# Let user set shortname and preferred name in signup flow

## 🌟 What is the purpose of this PR?




This PR re-introduces the functionality of the "Account Setup" page, where the user is able to specify their `shortname` and `preferredName`.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1202621375435417/1202795672151786/f) _(internal)_

## 🔍 What does this change?




- remove `properties` object on the `User` and `Org` GQL types to make the properties accessible on the top-level, and not confuse why the definition of the object does not match the `properties` object of the corresponding entities in the graph.

## 📜 Does this require a change to the docs?



No.

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



- `user.model.test.ts`

## ❓ How to test this?



1. Checkout the branch / view the deployment
2. Run the hash workspace using `yarn external-services up` and `yarn dev`
3. Go to `localhost:3000/signup`, and create a new account
4. Enter a valid username and preferred name, and submit
5. You should be redirected to the homepage. You should be greeted with your preferred name, and your kratos identity should contain your username.

## 📹 Demo
